api/nodes/views.py

```diff
@@ -638,6 +635,19 @@ def perform_create(self, serializer):
         user = self.request.user
         serializer.save(creator=user, parent=self.get_node())
 
+    def get_serializer_context(self):
+        context = super(NodeChildrenList, self).get_serializer_context()
+        region_id = self.request.query_params.get('region', None)
+        if region_id:
+            try:
+                region = Region.objects.get(_id=region_id)
```

Comment:
Same here; you only need to load the `id` rather than get the entire region object.